packages/server/src/api/controllers/role.ts

```diff
@@ -25,43 +31,35 @@ async function updateRolesOnUserTable(
   db: Database,
   roleId: string,
   updateOption: string,
-  roleVersion: string | undefined
+  roleVersion?: string
 ) {
   const table = await sdk.tables.getTable(InternalTables.USER_METADATA)
-  const schema = table.schema
-  const remove = updateOption === UpdateRolesOptions.REMOVED
-  let updated = false
-  for (let prop of Object.keys(schema)) {
-    if (prop === "roleId") {
-      updated = true
-      const constraints = schema[prop].constraints!
-      const updatedRoleId =
-        roleVersion === roles.RoleIDVersion.NAME
-          ? roles.getExternalRoleID(roleId, roleVersion)
-          : roleId
-      const indexOfRoleId = constraints.inclusion!.indexOf(updatedRoleId)
-      if (remove && indexOfRoleId !== -1) {
-        constraints.inclusion!.splice(indexOfRoleId, 1)
-      } else if (!remove && indexOfRoleId === -1) {
-        constraints.inclusion!.push(updatedRoleId)
-      }
-      break
+  const constraints = table.schema.roleId?.constraints
+  if (constraints) {
```

Comment:
NAB: Could reduce nesting with a guard statement/early return?

Reply:
Really good call. I'm a big fan of early returns / minimised nesting. Agree with basically every point in this video: https://www.youtube.com/watch?v=CFRhGnuXG-4.

Reply:
Haha I think I've actually seen that video before.